addons/mail/models/mail_activity.py

```diff
@@ -802,15 +828,20 @@ def _search_activity_state(self, operator, value):
         )
         return [('id', 'not in' if reverse_search else 'in', [r[0] for r in self._cr.fetchall()])]
 
-    @api.depends('activity_ids.date_deadline')
+    @api.depends('activity_ids.date_deadline', 'activity_ids.tz')
```

Comment:
This is why we need new tz field, instead of using ``user_id.tz`` -- otherwise change user's timezone may trigger too much recomputations